src/Microsoft.OData.Core/ODataInnerError.cs

```diff
@@ -93,5 +133,34 @@ internal string ToJson()
                 this.StackTrace == null ? "" : JsonValueUtils.GetEscapedJsonString(this.StackTrace),
                 this.InnerError == null ? "{}" : this.InnerError.ToJson());
         }
-}
+
+        private string LookUpStringValue(string v)
+        {
+            if (Properties == null)
+            {
+                return string.Empty;
+            }
+
+            foreach (KeyValuePair<string, ODataValue> pair in Properties)
+            {
+                if (pair.Key == v)
+                {
+                    string value = (string)pair.Value.FromODataValue();
+                    return value;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private void SetStringValue(string v, string value)
+        {
+            if (Properties == null)
+            {
+                Properties = new Dictionary<string, ODataValue>();
+            }
+
+            Properties[v] = value.ToODataValue();

```

Comment:
>Properties[v] [](start = 12, length = 13)

shall we check "v" is duplicated?

Reply:
We should overwrite the value. Remember, we are trying to mimic normal properties. 

---
In reply to: [320879295](https://github.com/OData/odata.net/pull/1523#discussion_r320879295) [](ancestors = 320879295)

Reply:
:thumbsup:

---
In reply to: [320930828](https://github.com/OData/odata.net/pull/1523#discussion_r320930828) [](ancestors = 320930828,320879295)